{{ cookiecutter.package_name }}/.github/workflows/ci.yml

```diff
@@ -90,21 +90,17 @@ jobs:
     secrets:
       CODECOV_TOKEN: {{ '${{ secrets.CODECOV_TOKEN }}' }}
 
-{% if cookiecutter.use_compiled_extensions == 'y' -%}
   publish:
-    # Build wheels when pushing to any branch except main
-    # publish.yml will only publish if tagged ^v.*
+    needs: [test, docs]
+    # Build wheels on PRs only when labelled. Releases will only be published if tagged ^v.*
+    # see https://github-actions-workflows.openastronomy.org/en/latest/publish.html#upload-to-pypi
     if: |
+      github.event_name != 'pull_request' ||
       (
-        github.event_name != 'pull_request' && (
-          github.ref_name != 'main' ||
-          github.event_name == 'workflow_dispatch'
-        )
-      ) || (
         github.event_name == 'pull_request' &&
         contains(github.event.pull_request.labels.*.name, 'Run publish')
       )
-    needs: [test, docs]
+{% if cookiecutter.use_compiled_extensions == 'y' -%}
```

Comment:
@CyclingNinja We need to double check that moving some of this out of the if statement hasn't broken the whitespace.

Reply:
It had, but I fixed it with the new test stuff.